main.go

```diff
@@ -1,7 +1,328 @@
 package main
 
-import "fmt"
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"time"
 
+	jwt "github.com/vladimiroff/jwt-go"
+
+	"github.com/gorilla/mux"
+)
+
+// Certificate Struct
+type Certificate struct {
+	ID        string    `json:"id"`
+	Title     string    `json:"title"`
+	CreatedAt string    `json:"createdAt"` // Will be a type date
+	OwnerID   string    `json:"ownerId"`
+	Year      int       `json:"year"`
+	Note      string    `json:"note"`
+	Transfer  *Transfer `json:"transfer"`
+}
+
+// User Struct
+type User struct {
+	ID    string `json:"id"`
+	Email string `json:"email"`
+	Name  string `json:"name"`
+}
+
+// Transfer Struct
+type Transfer struct {
+	To     string `json:"to"`
+	Status string `json:"status"`
+}
+
+// In memory data
+var certificates []Certificate              //Certificates
+var users []User                            //Users
+var mySigningKey = []byte("mockSigningKey") //mock signing key
+
+//Homehandler for basic testing
+func homeHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode("HOMEPAGE")
+}
+
+// Initiate some in memory data
+func initInMemoryData() {
+	userA := User{
```

Comment:
We will want to delete this in memory init data and put it into unit tests. As a test we setup some fake data then call the handlers end point with that data and ensure we get the expected results back. Since the data in is always the same it allows us to have an automated way of ensuring our end points don't break. As our unit tests would fail and we could catch it before deploying.